pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -618,31 +621,100 @@ func (c *Cluster) saveMonConfig() error {
+		return fmt.Errorf("failed to make mon %s pvc. %+v", d.Name, err)
+	}
+	var pvcExists bool
+	if deploymentExists {
+		_, err = c.context.Clientset.CoreV1().PersistentVolumeClaims(
+			c.Namespace).Get(pvc.Name, metav1.GetOptions{})
+		if err == nil {
+			pvcExists = true
+		} else if errors.IsNotFound(err) {
+			pvcExists = false
+		} else {
+			return fmt.Errorf("failed to get pvc %s. %+v", pvc.Name, err)
 		}
-		logger.Infof("deployment for mon %s already exists. updating if needed", m.ResourceName)
+	} else {
+		pvcExists = false
+	}
+
+	// persistent storage is not altered after the deployment is created. this
+	// means we need to be careful when updating the deployment to avoid new
+	// changes to the crd to change an existing pod's persistent storage. the
+	// deployment spec created above does not specify persistent storage. here
+	// we add in PVC or HostPath storage based on an existing deployment OR on
+	// the current state of the CRD.
+	if pvcExists || (!deploymentExists && c.spec.Mon.VolumeClaimTemplate != nil) {
+		d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes,
+			opspec.DaemonVolumesDataPVC(pvc.Name))
+	} else {
+		d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes,
+			opspec.DaemonVolumesDataHostPath(m.DataPathMap)...)
+	}
+
+	if deploymentExists {
+		logger.Infof("deployment for mon %s already exists. updating if needed",
+			d.Name)
 
-		// Always invoke ceph version before an upgrade so we are sure to be up-to-date
 		daemonType := string(config.MonType)
 		var cephVersionToUse cephver.CephVersion
-		currentCephVersion, err := client.LeastUptodateDaemonVersion(c.context, c.clusterInfo.Name, daemonType)
+		currentCephVersion, err := client.LeastUptodateDaemonVersion(c.context,
+			c.clusterInfo.Name, daemonType)
 		if err != nil {
-			logger.Warningf("failed to retrieve current ceph %s version. %+v", daemonType, err)
-			logger.Debug("could not detect ceph version during update, this is likely an initial bootstrap, proceeding with c.clusterInfo.CephVersion")
+			logger.Warningf("failed to retrieve current ceph %s version. %+v",
+				daemonType, err)
+			logger.Debug("could not detect ceph version during update, this " +
+				"is likely an initial bootstrap, proceeding with " +
+				"c.clusterInfo.CephVersion")
 			cephVersionToUse = c.clusterInfo.CephVersion
 		} else {
-			logger.Debugf("current cluster version for monitors before upgrading is: %+v", currentCephVersion)
+			logger.Debugf("current cluster version for monitors before "+
+				"upgrading is: %+v", currentCephVersion)
 			cephVersionToUse = currentCephVersion
 		}
-		if err := updateDeploymentAndWait(c.context, d, c.Namespace, daemonType, m.DaemonName, cephVersionToUse); err != nil {
-			return fmt.Errorf("failed to update mon deployment %s. %+v", m.ResourceName, err)
+
+		// enable updates again when we figure out how to test for pvc changes
```

Comment:
does this comment still apply? looks like updates are already enabled

Reply:
ahh... if i hadn't reformatted all those strings, i'd probably see that in the diff :P